### ISSUE TITLE
UndeliverableException on com.instabug.library.network.NetworkManager

### ISSUE THREAD

**Issue by IgorMaineti** [Instabug/Instabug-Android#330]
### Steps to Reproduce the Problem

No specific steps to reproduce this issue.

### Expected Behavior

The app shouldn't crash.

### Actual Behavior

The app is crashing with the following stack trace:

`Fatal Exception: io.reactivex.exceptions.UndeliverableException: java.net.UnknownHostException: Unable to resolve host "api.instabug.com": No address associated with hostname
       at io.reactivex.plugins.RxJavaPlugins.onError(RxJavaPlugins.java:367)
       at io.reactivex.internal.operators.completable.CompletableMergeIterable$MergeCompletableObserver.onError(CompletableMergeIterable.java:124)
       at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.onError(CompletableSubscribeOn.java:74)
       at io.reactivex.internal.operators.completable.CompletableConcatArray$ConcatInnerObserver.onError(CompletableConcatArray.java:60)
       at io.reactivex.internal.operators.completable.CompletableConcatArray$ConcatInnerObserver.onError(CompletableConcatArray.java:60)
       at io.reactivex.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onError(CompletablePeek.java:96)
       at io.reactivex.internal.operators.observable.ObservableIgnoreElementsCompletable$IgnoreObservable.onError(ObservableIgnoreElementsCompletable.java:61)
       at io.reactivex.internal.operators.observable.ObservableRetryPredicate$RepeatObserver.onError(ObservableRetryPredicate.java:89)
       at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.tryOnError(ObservableCreate.java:85)
       at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onError(ObservableCreate.java:73)
       at com.instabug.library.network.NetworkManager$a.subscribe(NetworkManager.java:104)
       at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
       at io.reactivex.Observable.subscribe(Observable.java:12025)
       at io.reactivex.internal.operators.observable.ObservableRetryPredicate$RepeatObserver.subscribeNext(ObservableRetryPredicate.java:111)
       at io.reactivex.internal.operators.observable.ObservableRetryPredicate.subscribeActual(ObservableRetryPredicate.java:41)
       at io.reactivex.Observable.subscribe(Observable.java:12025)
       at io.reactivex.internal.operators.observable.ObservableIgnoreElementsCompletable.subscribeActual(ObservableIgnoreElementsCompletable.java:31)
       at io.reactivex.Completable.subscribe(Completable.java:1919)
       at io.reactivex.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
       at io.reactivex.Completable.subscribe(Completable.java:1919)
       at io.reactivex.internal.operators.completable.CompletableConcatArray$ConcatInnerObserver.next(CompletableConcatArray.java:89)
       at io.reactivex.internal.operators.completable.CompletableConcatArray.subscribeActual(CompletableConcatArray.java:33)
       at io.reactivex.Completable.subscribe(Completable.java:1919)
       at io.reactivex.internal.operators.completable.CompletableConcatArray$ConcatInnerObserver.next(CompletableConcatArray.java:89)
       at io.reactivex.internal.operators.completable.CompletableConcatArray.subscribeActual(CompletableConcatArray.java:33)
       at io.reactivex.Completable.subscribe(Completable.java:1919)
       at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
       at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:571)
       at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
       at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)`

### Instabug integration code
`Instabug.Builder(this, INSTABUG_APP_TOKEN)
            .setInvocationEvents(InstabugInvocationEvent.SCREENSHOT)
            .build()
        Instabug.setWelcomeMessageState(WelcomeMessage.State.DISABLED)`

### SDK Version
9.1.6

### Android Version
10

### Device Model
Brand: Xiaomi
Model: MI 9 SE
Orientation: Portrait

**Comment by MahmoudAmin39** [comment]
Hi @IgorMaineti, Thanks for reporting this.
We created a **SNAPSHOT** that should fix the issue, Can you please try it and tell us if it fixed the issue for you?
```
implementation 'com.instabug.library:instabug:9.1.6.2716-SNAPSHOT'
```
you will need to add these lines to your app **build.gradle** before your dependencies
```
repositories {
    ....
    maven {
        url "https://oss.sonatype.org/content/repositories/snapshots"
    }
}
```
This **SNAPSHOT** is completely safe to use and has no debug code at all.

Thank You :smiley:

**Comment by IgorMaineti** [comment]
Hello @MahmoudAmin39, I will wait for the next release to test if all crashes were resolved haha, thanks.

**Comment by Muhammadyoussef** [comment]
Hi @IgorMaineti,

This issue has been fixed in our major release `v10.0.0` 🎉
